### PR TITLE
chore: Using Temp DB for Delete Sample Tests

### PR DESCRIPTION
**Pull request by anuragsrivstv**
Parent Issue #2093

-  This is 1st PR containing changes to use temporary databases for all sample tests that modify the database. 

- This PR contains All Delete Sample Tests that needs to be updated.

- Number of affected tests files 3 out of 21 in closed PR #2216

### CODE REVIEW

**Comment by anuragsrivstv** [comment]
Hi @amanda-tarafa , Can you please have a look at this PR. Thanks !

**Comment by amanda-tarafa** [comment]
Yes, we'll do as soon as I get a chance, thanks!

File: spanner/api/Spanner.Samples.Tests/DeleteUsingPartitionedDmlCoreAsyncTest.cs [hunk 1]
```diff
@@ -28,8 +29,30 @@ public DeleteUsingPartitionedDmlCoreAsyncTest(SpannerFixture spannerFixture)
...
     {
-        DeleteUsingPartitionedDmlCoreAsyncSample sample = new DeleteUsingPartitionedDmlCoreAsyncSample();
-        var rowCount = await sample.DeleteUsingPartitionedDmlCoreAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, _spannerFixture.DatabaseId);
-        Assert.True(rowCount >= 1);
+        await _spannerFixture.RunWithTemporaryPostgresDatabaseAsync(async databaseId =>
+        {
+            await InsertDataAsync(databaseId);
+            DeleteUsingPartitionedDmlCoreAsyncSample sample = new DeleteUsingPartitionedDmlCoreAsyncSample();
+            var rowCount = await sample.DeleteUsingPartitionedDmlCoreAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, databaseId);
+            Assert.True(rowCount >= 1);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/DeleteUsingPartitionedDmlCoreAsyncTest.cs:37]: I think that now we probably know exactly how many rows were deleted.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]

File: spanner/api/Spanner.Samples.Tests/DeleteUsingDmlCoreAsyncTest.cs [hunk 2]
```diff
@@ -28,8 +28,12 @@ public DeleteUsingDmlCoreAsyncTest(SpannerFixture spannerFixture)
     [Fact]
     public async Task TestDeleteUsingDmlCoreAsync()
     {
-        DeleteUsingDmlCoreAsyncSample sample = new DeleteUsingDmlCoreAsyncSample();
-        var rowCount = await sample.DeleteUsingDmlCoreAsync(_spannerFixture.ProjectId, _spannerFixture.InstanceId, _spannerFixture.DatabaseId);
-        Assert.Equal(1, rowCount);
+        await _spannerFixture.RunWithTemporaryDatabaseAsync(async databaseId =>
+        {
+            await _spannerFixture.CreateSingersTableAndInsertDataAsync(databaseId);
```
[the remark this answers is not in the document]
Comment (anuragsrivstv) [reply spanner/api/Spanner.Samples.Tests/DeleteUsingDmlCoreAsyncTest.cs:33]: Yes, in SpannerFixture.InitializeDatabaseAsync()

File: spanner/api/Spanner.Samples.Tests/DeleteUsingDmlReturningAsyncPostgresTest.cs [hunk 3]
```diff
@@ -30,18 +30,22 @@ public DeleteUsingDmlReturningAsyncPostgresTest(SpannerFixture spannerFixture)
     [Fact]
     public async Task TestDeleteUsingDmlReturningAsyncPostgres()
     {
-        await InsertDataAsync();
-        var sample = new DeleteUsingDmlReturningAsyncPostgresSample();
-        var deletedSingerNames = await sample.DeleteUsingDmlReturningAsyncPostgres(_spannerFixture.ProjectId, _spannerFixture.InstanceId, _spannerFixture.PostgreSqlDatabaseId);
+        await _spannerFixture.RunWithTemporaryPostgresDatabaseAsync(async databaseId =>
+        {
+            await InsertDataAsync(databaseId);
```
[the remark this answers is not in the document]
Comment (anuragsrivstv) [reply spanner/api/Spanner.Samples.Tests/DeleteUsingDmlReturningAsyncPostgresTest.cs:35]: >The version of RunWithTemporaryDatabaseAsync for non-postgress Database has some overloads that take extra statements, can we do similarly for Postgress, mainly for uniformity sake.
There cannot be Extra DDL statements while creating PostgreSQL.

>And also, there are some methods in the fixture called CreateSingersTableAndInsertData etc. Couldn't we just use the same methods everywhere? Why do we need this particular row to test that delete works?
CreateSingersTableAndInsertData Create the singers table and its containes DDL statement in Google Standard SQL format. Therefor we can not use the same function for Postgres. Also I believe going forward it's better to insert data there itself for tests using Temp DB as that way tests can be completely independent.

**Comment by anuragsrivstv** [comment]
Tests are failing right now, will fix it.

**Comment by anuragsrivstv** [comment]
Please do not review it right  now. Spanner fixture files has some conflicts. Resolving it.

**Comment by anuragsrivstv** [comment]
Closing this PR as it's outdated. Will raise another one instead of it.

**Pull request closed without merging**